examples/control-vector-generator/control-vector-generator.cpp

```diff
@@ -0,0 +1,648 @@
+        const bool add_bos = llama_should_add_bos_token(llama_get_model(ctx));
+        tokens_pos = ::llama_tokenize(ctx, pos, add_bos);
+        tokens_neg = ::llama_tokenize(ctx, neg, add_bos);
+        max_seq_len = std::max(tokens_pos.size(), tokens_neg.size());
+        padding_seq(ctx, tokens_pos, max_seq_len);
+        padding_seq(ctx, tokens_neg, max_seq_len);
+    }
+
+    void padding_seq(llama_context * ctx, std::vector<llama_token> & tokens, size_t len) {
+        // TODO: customize padding token
+        std::vector<llama_token> pad_tokens = ::llama_tokenize(ctx, " ", false);
+        llama_token pad_tok = pad_tokens.back();
+        while (tokens.size() < len) {
+            tokens.push_back(pad_tok);
+        }
+    }
+};
+
+//////////////////////////////////////////////////
+
+template <typename T>
+static std::string to_string(const T & val) {
+    std::stringstream ss;
+    ss << val;
+    return ss.str();
+}
+
+static void print_usage(const char * executable) {
+    struct ctrl_params defaults;
+    printf("\n");
+    printf("usage: %s [options] -m <model> [gpt-opts]", executable);
+    printf("\n");
+    printf("Creates a GGUF control vector for a given model.");
+    printf("\n");
+    printf("options:\n");
+    printf("  -h,  --help                 show this help message and exit\n");
+    printf("  -o,  --outfile FNAME        output file\n");
+    printf("                                default: %s\n", defaults.outfile.c_str());
+    printf("  -pf, --positive-file FNAME  positive prompts file, one prompt per line\n");
+    printf("                                default: %s\n", defaults.positive_prompts_file.c_str());
+    printf("  -nf, --negative-file FNAME  negative prompts file, one prompt per line\n");
+    printf("                                default: %s\n", defaults.negative_prompts_file.c_str());
+    printf("  -cf, --completions-file     completions file\n");
+    printf("                                default: %s\n", defaults.completions_file.c_str());
+    printf("  -nc, --num-completions N    number of lines of completions file to use\n");
+    printf("                                default: %d\n", defaults.n_completions);
+    printf("  --batch-pca N               batch size used for PCA. Larger batch runs faster, but uses more memory\n");
+    printf("                                default: %d\n", defaults.n_pca_batch);
+    printf("  --iter-pca N                number of iterations used for PCA\n");
+    printf("                                default: %d\n", defaults.n_pca_iterations);
+    printf("\n");
+    printf("gpt-opts:\n");
+    printf("  -m, --model  FNAME          path to model file\n");
+    printf("  -ngl,  --gpu-layers N       number of layers to offload to GPU\n");
+    printf("  ...other options from main\n");
+    printf("\n");
+}
+
+static int ctrlvec_params_parse_ex(int argc, char ** argv, ctrl_params & params) {
```

Comment:
Let's merge `ctrl_params` into `gpt_params` so that we have a consistent handling of CLI args in all examples

Reply:
I didn't noticed that the `gpt_params` has been refactored. It's way easier to work with it now!

I moved `ctrl_params` to `gpt_params`. Please have a look on https://github.com/ggerganov/llama.cpp/pull/7514/commits/679f5137f88ebfbc1aa2e8160cfe0c9633982e21 . Thanks!